### PR TITLE
Assertions for two-dimensional arrays of char

### PR DESCRIPTION
**Pull request by mwajcht**
#### Check List:
* Fixes #1182 
* Unit tests : YES
* Javadoc with a code example (on API only) : YES

### CODE REVIEW

File: src/main/java/org/assertj/core/api/Char2DArrayAssert.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  public Char2DArrayAssert doesNotContain(char[] value, Index index) {
+    arrays.assertDoesNotContain(info, actual, value, index);
+    return myself;
+  }
+
+  /**
+   * Use unicode character representation instead of standard representation in error messages.
+   * <p>
+   * With standard error message:
+   * <pre><code class='java'> assertThat("a6c".toCharArray()).isEqualTo("abó".toCharArray());
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/Char2DArrayAssert.java:157]: examples don't show char[][] which is a bit confusing

File: src/main/java/org/assertj/core/api/InstanceOfAssertFactories.java [hunk 2]
```diff
@@ -263,6 +263,12 @@
   InstanceOfAssertFactory<char[], AbstractCharArrayAssert<?>> CHAR_ARRAY = new InstanceOfAssertFactory<>(char[].class,
                                                                                                          Assertions::assertThat);
 
+  /**
+   * {@link InstanceOfAssertFactory} for a {@code char} two-dimensional array.
+   */
+  InstanceOfAssertFactory<char[][], Char2DArrayAssert> CHAR_2D_ARRAY = new InstanceOfAssertFactory<>(char[][].class,
+                                                                                                         Assertions::assertThat);
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/InstanceOfAssertFactories.java:270]: formatting is off, Assertions::assertThat should be aligned with the first param, make sure your IDE follows our formatting convention

File: src/test/java/org/assertj/core/api/char2darray/Char2DArrayAssert_isDeepEqualTo_Test.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+@DisplayName("Char2DArrayAssert isDeepEqualTo")
+class Char2DArrayAssert_isDeepEqualTo_Test {
+
+  @Test
+  void should_pass_if_both_actual_and_expected_are_null() {
+    // GIVEN
+    char[][] actual = null;
+    char[][] expected = null;
+    // WHEN/THEN
+    assertThat(actual).isDeepEqualTo(expected);
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/char2darray/Char2DArrayAssert_isDeepEqualTo_Test.java:43]: replace by `assertThat` by `then` (from BDDAssertions) in all tests

File: src/test/java/org/assertj/core/internal/char2darrays/Char2DArrays_assertDoesNotContain_at_Index_Test.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  }
+
+  @Test
+  public void should_throw_error_if_Index_is_null() {
+    // GIVEN
+    Index nullIndex = null;
+    // WHEN/THEN
+    assertThatNullPointerException().isThrownBy(() -> arrays.assertDoesNotContain(someInfo(), actual, new char[] { 'a', 'b', 'c' },
+      nullIndex))
+      .withMessage("Index should not be null");
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/internal/char2darrays/Char2DArrays_assertDoesNotContain_at_Index_Test.java:67]: formatting: .withMessage should be aligned with .isThrownBy

File: src/main/java/org/assertj/core/api/Char2DArrayAssert.java (hunk 1 above)
Comment (mwajcht) [reply src/main/java/org/assertj/core/api/Char2DArrayAssert.java]: done

File: src/main/java/org/assertj/core/api/InstanceOfAssertFactories.java (hunk 2 above)
Comment (mwajcht) [reply src/main/java/org/assertj/core/api/InstanceOfAssertFactories.java]: Sorry, I had different formatter set for this project. Fixed thant and reformatted the code with formatter used in AssertJ.

File: src/test/java/org/assertj/core/api/char2darray/Char2DArrayAssert_isDeepEqualTo_Test.java (hunk 3 above)
Comment (mwajcht) [reply src/test/java/org/assertj/core/api/char2darray/Char2DArrayAssert_isDeepEqualTo_Test.java]: done

File: src/test/java/org/assertj/core/internal/char2darrays/Char2DArrays_assertDoesNotContain_at_Index_Test.java (hunk 4 above)
Comment (mwajcht) [reply src/test/java/org/assertj/core/internal/char2darrays/Char2DArrays_assertDoesNotContain_at_Index_Test.java]: done

**Comment by joel-costigliola** [comment]
Integrated thanks @mwajcht!

**Pull request closed without merging**